cmd/zoekt-sourcegraph-indexserver/main.go

```diff
@@ -241,6 +254,23 @@ func (s *Server) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	repoTmpl.Execute(w, data)
 }
 
+func (s *Server) updateOptions(w http.ResponseWriter, r *http.Request) {
+	if r.Method != "POST" {
+		http.Error(w, "method not allowed", http.StatusMethodNotAllowed)
+		return
+	}
+	err := r.ParseForm()
+	if err != nil {
+		http.Error(w, fmt.Sprintf("error parsing body: %v", err), http.StatusBadRequest)
+		return
+	}
+	if lf, ok := r.Form["large_files"]; ok {
+		s.optsMu.Lock()
+		s.LargeFiles = lf[0]
+		s.optsMu.Unlock()
+	}
```

Comment:
you may find that just passing JSON messages is more straightforward than trying to do a fully accurate REST API, i.e. instead of using form data. We do this elsewhere and imo it works quite nicely. i.e. everything is just POST with JSON bodies, get a JSON response back and you can neglect all other details of HTTP (response status code indicates a server failure, never an operational failure, etc.)

Reply:
I went with a form body to be consistent with other data being sent to this service. See the rest of the `ServeHTTP` function for the other form data. I'm fine with changing it to JSON but would like to hear @keegancsmith 's opinion on the consistency here.